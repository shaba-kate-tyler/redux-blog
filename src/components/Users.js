import React from 'react';
import PropTypes from 'prop-types';
import User from './User';

function Users({ users }) {
  const userList = users.map(user => {
    return <li key={user.id}>{<User { ...user }/>}</li>;
  });
  return (
    <ul>
      {userList}
    </ul>
  );
}

Users.propTypes = {
  users: PropTypes.array.isRequired
};

export default Users;
