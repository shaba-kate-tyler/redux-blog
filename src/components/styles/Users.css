.Users {
    background-color: rgb(244, 244, 244);
    margin: 2vh;
    align-items: center;
    list-style: none;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    li:hover {
        background-color: rgba(255, 255, 114, 0.2);
        transition: 0.2s;
        cursor: pointer;
    }
    li {
        margin: 4px;
        border: 1px solid black;
        border-radius: 7px;
    }
}