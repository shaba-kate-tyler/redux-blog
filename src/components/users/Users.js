import CreateList from '../higher-order/CreateList';
import PropTypes from 'prop-types';
import User from './User';
import styles from '../styles/Users.css';

function Users({ users }) {
  return CreateList(User, users, styles.Users);
}

Users.propTypes = {
  users: PropTypes.array.isRequired
};

export default Users;
