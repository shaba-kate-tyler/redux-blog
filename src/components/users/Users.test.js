import React from 'react';
import Users from './Users';
import renderer from 'react-test-renderer';
// import User from './User';

describe('Users', () => {
  it('matches a snapshot', () => {
    const props = {
      users: [
        {
          id: 1,
          name: 'kaiya',
          username: 'ladybeard',
          website: 'schnepherds4life.org',
          email: 'itsme@kai.com'
        },
        {
          id: 2,
          name: 'kingsley',
          username: 'fathersLittleTiger',
          website: 'carolKibble.org',
          email: 'yasssss@kitty.com'
        },
        {
          id: 3,
          name: 'tyler',
          username: 'tylerCorbett',
          website: 'tylerTheDev.org',
          email: 'sayhi@tyler.com'
        }
      ]
    };

    const tree = renderer.create(
      <Users { ...props } />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
