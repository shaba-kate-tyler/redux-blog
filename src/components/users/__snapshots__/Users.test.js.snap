// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users matches a snapshot 1`] = `
<ul>
  <li>
    <a
      href="/users/1"
      onClick={[Function]}
    >
      <div>
        <h3>
          kaiya
        </h3>
        <h4>
          ladybeard
        </h4>
        <p>
          schnepherds4life.org
           / 
          itsme@kai.com
        </p>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/users/2"
      onClick={[Function]}
    >
      <div>
        <h3>
          kingsley
        </h3>
        <h4>
          fathersLittleTiger
        </h4>
        <p>
          carolKibble.org
           / 
          yasssss@kitty.com
        </p>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/users/3"
      onClick={[Function]}
    >
      <div>
        <h3>
          tyler
        </h3>
        <h4>
          tylerCorbett
        </h4>
        <p>
          tylerTheDev.org
           / 
          sayhi@tyler.com
        </p>
      </div>
    </a>
  </li>
</ul>
`;
